File: AdvancedMap.Objective-C/AdvancedMap/Languages.m
//
//  Languages.m
//  AdvancedMap.Objective-C
//
//  Created by Aare Undo on 22/09/2017.
//  Copyright © 2017 Nutiteq. All rights reserved.
//

#import "Languages.h"

@implementation Language

- (id)initWithName:(NSString *)name value: (NSString *)value {
    self.name = name;
    self.value = value;
    
    return self;
}

@end

@implementation Languages

+ (NSMutableArray *)getList {
    static NSMutableArray *list = nil;
    
    if (list == nil) {
        
        list = [[NSMutableArray alloc] init];
        
        [list addObject:[[Language alloc] initWithName:@"Local" value:@""]];
        [list addObject:[[Language alloc] initWithName:@"English" value:@"en"]];
        [list addObject:[[Language alloc] initWithName:@"German" value:@"de"]];
        [list addObject:[[Language alloc] initWithName:@"Spanish" value:@"es"]];
        [list addObject:[[Language alloc] initWithName:@"Italian" value:@"it"]];
        [list addObject:[[Language alloc] initWithName:@"French" value:@"fr"]];
        [list addObject:[[Language alloc] initWithName:@"Russian" value:@"ru"]];
    }
    
    return list;
}

@end

File: AdvancedMap.Swift/Feature Demo/Languages.swift
//
//  Languages.swift
//  Feature Demo
//
//  Created by Aare Undo on 20/06/2017.
//  Copyright © 2017 CARTO. All rights reserved.
//

import Foundation
import UIKit

class Languages {
    
    static var list = [Language]()
    
    static func initialize() {
        
        var language = Language()
        language.name = "Default"
        language.value = ""
        
        list.append(language)
        
        language = Language()
        language.name = "English"
        language.value = "en"
        
        list.append(language)
        
        language = Language()
        language.name = "German"
        language.value = "de"
        
        list.append(language)
        
        language = Language()
        language.name = "Spanish"
        language.value = "es"
        
        list.append(language)
        
        language = Language()
        language.name = "Italian"
        language.value = "it"
        
        list.append(language)
        
        language = Language()
        language.name = "French"
        language.value = "fr"
        
        list.append(language)
        
        language = Language()
        language.name = "Russian"
        language.value = "ru"
        
        list.append(language)
    }
}

class Language : NSObject {
    
    var name: String!
    
    var value: String!
}





